core/src/main/java/org/elasticsearch/index/translog/TranslogWriter.java

```diff
@@ -192,7 +192,18 @@ public synchronized TranslogReader closeIntoReader() throws IOException {
             throw e;
         }
         if (closed.compareAndSet(false, true)) {
-            return new TranslogReader(generation, channel, path, firstOperationOffset, getWrittenOffset(), operationCounter);
+            logger.trace("closing into a reader. path [{}]", path);
+            try {
+                return new TranslogReader(generation, channel, path, firstOperationOffset, getWrittenOffset(), operationCounter, shardId);
+            } catch (Throwable t) {
+                // close the channel, as we are closed and failed to create a new reader
+                try {
+                    channel.close();
```

Comment:
there is a valid recent criticism about it swallowing exceptions and I wanted see where things come from, but yeah, can change it here and wait for the bigger change on closeWhileHandlingExceptions semantics